cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -355,4 +360,14 @@ public void execute(TxRunnable runnable) throws TransactionFailureException {
   public void execute(int timeoutInSeconds, TxRunnable runnable) throws TransactionFailureException {
     transactional.execute(timeoutInSeconds, runnable);
   }
+
+  @Override
+  public DataTracer getDataTracer(String dataTracerName) {
+    return dataTracerFactory.getTracer(dataTracerName, program.getId().getParent());
+  }
+
+  @Inject(optional = true)
+  private void setDataTracerFactory(@Named("DataTracerFactory") DataTracerFactory dataTracerFactory) {
```

Comment:
You don't need to call this method and the Guice framework will call it. However, you shouldn't have the `@Named`, unless you have different bindings for the `DataTracerFactory` type.